ui/packages/catalog-ui-search/src/main/webapp/component/search-form/forms-sharing/search-form-sharing.collection.js

```diff
@@ -12,109 +12,112 @@
-                        id: value.id,
-                        name: value.title,
-                        description: value.description,
-                        type: 'custom',
-                        filterTemplate: JSON.stringify(value.filterTemplate),
-                        accessIndividuals: value.accessIndividuals,
-                        accessGroups: value.accessGroups,
-                        createdBy: value.creator,
-                        owner: value.owner,
-                        querySettings: value.querySettings
-                    }));
-                }
-            });
-            this.doneLoading();
+module.exports = Backbone.AssociatedModel.extend({
+  model: SearchForm,
+  defaults: {
+    doneLoading: false,
+    sharedSearchForms: []
+  },
+  initialize: function () {
+    this.addMySharedForms()
+  },
+  relations: [{
+    type: Backbone.Many,
+    key: 'sharedSearchForms',
+    collectionType: Backbone.Collection.extend({
+      model: SearchForm,
+      initialize: function () {}
+    })
+  }],
+  addMySharedForms: function () {
+    if (!this.isDestroyed) {
+      if (promiseIsResolved === true) {
+        promiseIsResolved = false
+        bootstrapPromise = new sharedSearchFormPromise()
+      }
+      bootstrapPromise.then(() => {
+        $.each(sharedTemplates, (index, value) => {
+          if (this.checkIfShareable(value)) {
+            let utcSeconds = value.created / 1000
+            let d = new Date(0)
+            d.setUTCSeconds(utcSeconds)
+            this.addSearchForm(new SearchForm({
+              createdOn: Common.getHumanReadableDate(d),
+              id: value.id,
+              name: value.title,
+              description: value.description,
+              type: 'custom',
+              filterTemplate: JSON.stringify(value.filterTemplate),
+              accessIndividuals: value.accessIndividuals,
+              accessGroups: value.accessGroups,
+              createdBy: value.creator,
+              owner: value.owner,
+              querySettings: value.querySettings
+            }))
+          }
         })
+        this.doneLoading()
+      })
     };
-   },
-   checkIfShareable: function(template) {
-       if (this.checkIfInGroup(template) || this.checkIfInIndividiuals(template)) {
-           return true;
-       }
-       return false;
-   },
-   checkIfInGroup: function(template) {
-       let myGroups = user.get('user').get('roles');
-       let roleIntersection = myGroups.filter(function(n) {
-        return template.accessGroups.indexOf(n) !== -1;
-       });
+  },
+  checkIfShareable: function (template) {
+    if ((this.checkIfInGroup(template) || this.checkIfInIndividiuals(template)) && !this.checkIfOwner(template)) {
+      return true
+    }
+    return false
+  },
+  checkIfOwner: function (template) {
+    return user.get('user').get('userid') === template.owner
+  },
+  checkIfInGroup: function (template) {
+    let myGroups = user.get('user').get('roles')
```

Comment:
✏️ `myGroups` is never reassigned, might be worth considering making that official with `const` 😄